src/main/java/mods/railcraft/common/blocks/wayobjects/RoutingLogic.java

```diff
@@ -204,8 +222,8 @@ public NOT(Condition a) {
         }
 
         @Override
-        public boolean matches(IRoutingTile tile, EntityMinecart cart) {
-            return !a.matches(tile, cart);
+        public int evaluate(IRoutingTile tile, EntityMinecart cart) {
+            return valueForBool(a.evaluate(tile, cart) == 0);
```

Comment:
In that case, I would also clamp constants to [NO_POWER, FULL_POWER] and raise `constant.invalid` outside that range. I'm not familiar with the redstone-mod landscape--are there any that might benefit from RoutingLogic technically being able to emit a higher power than 15?